File: easy/bubble-sort/bubble-sort.js
const bubbleSort = (array) => {
    let swapped = false,
        i = 0;
    while (i < array.length-1) {
        if (array[i] > array[i+1]) {
            [ array[i], array[i+1] ] = [ array[i+1], array[i] ];
            swapped = true;
        } i++;
    }
    return swapped ? bubbleSort(array) : array
}

// General idea with bubble sort is to "bubble up" the larger numbers 

File: easy/bubble-sort/bubble-sort.py
# I: An array of ints
# O: a sorted array
# C: O(n^2) time | O(1) space
# E: N/A

def bubbleSort(array):
    swapped, i = False, 0
    while i < len(array)-1:
        if array[i] > array[i+1]:
            array[i], array[i+1], swapped = array[i+1], array[i], True
        i += 1
    if swapped == True:
        return bubbleSort(array)
    return array

